### PR TITLE
Added version information upon startup

### PR DESCRIPTION
**Pull request by siredmar**
Added some versioning information upon startup to help with issues like #48 
The output does look like this:
`$ ./julius`
`INFO: Built with git commit hash: v1.0.0-9-g7868516950-dirty`
`INFO: Initializing SDL`
`INFO: SDL initialized  `

There will be a generated version.h in src/platform/version.h by cmake which.

### CODE REVIEW

File: src/platform/julius.c [hunk 1]
```diff
@@ -325,9 +326,11 @@ static void teardown(void)
 
 int main(int argc, char **argv)
 {
+    SDL_Log("Built with git commit: %s\n", GIT_COMMIT_HASH);
+    
     const char *custom_data_dir = (argc > 1 && argv[1]) ? argv[1] : NULL;
     setup(custom_data_dir);
-
+
```
Comment (crudelios) [inline src/platform/julius.c]: Picky picky! Get rid of this extra space :P

Comment (siredmar) [reply src/platform/julius.c]: removed newline

File: src/platform/julius.c [hunk 2]
```diff
@@ -325,9 +326,10 @@ static void teardown(void)
 
 int main(int argc, char **argv)
 {
+    SDL_Log("Built with git commit: %s\n", GIT_COMMIT_HASH);
+    
     const char *custom_data_dir = (argc > 1 && argv[1]) ? argv[1] : NULL;
     setup(custom_data_dir);
-
```
Comment (bvschaik) [inline src/platform/julius.c]: Can I have the newline (excluding the spaces ;) ) back please?

Comment (siredmar) [reply src/platform/julius.c]: Done. You got the newline back.

**Comment by jdmillard** [comment]
I pulled `master` after this was merged. My console displays:
```
INFO: Built with git commit:
```

My "clean build process" on Windows is:
```
cd build
cmake.exe .. -G "MinGW Makefiles"
mingw32-make.exe -j8
julius.exe "path\to\GOG Games\Caesar 3"
```

...

OKAY, I took a closer look and this. This feature depends on cmake executing a git command which breaks for the following git setup: If "Git for Windows" is used to clone, then generic CMD is used to compile. Building this way might sound strange, but there are real reasons for some people to build this way on Windows machines (too hard to explain here).

I suggest to leave the new CMake logic as-is. However, if no commit hash string was set during CMakeLists.txt, the `INFO` print statement should be avoided altogether. Your thoughts?

**Pull request merged**